File: Medium/1014. Best Sightseeing Pair/sightseeingPair.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int maxScoreSightseeingPair(vector<int>& values) {
    int maxScore = 0;
    int maxI = values[0]; // Initialize with values[0] + 0

    for (int j = 1; j < values.size(); j++) {
        // Calculate the score for the current pair (i, j)
        maxScore = max(maxScore, maxI + values[j] - j);
        // Update maxI to include the current value and index
        maxI = max(maxI, values[j] + j);
    }

    return maxScore;
}

int main() {
    vector<int> values = {8, 1, 5, 2, 6};
    cout << maxScoreSightseeingPair(values) << endl; // Output: 11
    return 0;
}

File: Medium/1014. Best Sightseeing Pair/sightseeingPair.py
class Solution(object):
    def maxScoreSightseeingPair(self, values):
        """
        :type values: List[int]
        :rtype: int
        """
        max_score = 0
        max_i = values[0]  # Initialize with values[0] + 0
        
        for j in range(1, len(values)):
            # Calculate the score for the current pair (i, j)
            max_score = max(max_score, max_i + values[j] - j)
            # Update max_i to include the current value and index
            max_i = max(max_i, values[j] + j)
        
        return max_score

# Example usage
values = [8, 1, 5, 2, 6]
solution = Solution()
print(solution.maxScoreSightseeingPair(values))  # Output: 11